### DISCUSSION TITLE
how-to-set-up-lexical-editor-in-react-native

### DISCUSSION THREAD

**Discussion by giscus** [strdr4605/strdr4605.github.io#75] [General]
# how-to-set-up-lexical-editor-in-react-native

We will look at how to build a rich text editor in React Native using lexical editor, with the help of react-native-webview.

https://strdr4605.com/how-to-set-up-lexical-editor-in-react-native

**Comment by mariusplanable** [comment]
Great stuff, @strdr4605 🙌

**Comment by patr0cl0** [comment]
Hey, really good tutorial @strdr4605.

Quick question, woudn't it be possible to use the `injectedJavaScriptObject` prop from `WebView` in order to initialize the lexical state? Thanks

**Reply by patr0cl0** [reply]
Note: I tried to do it myself, but didn't have any luck

I did some test and couldn't make it work

```
<WebView
  ref={webviewRef}
  source={{ html: htmlString }}
  onMessage={onPreviewMesasge}
  javaScriptEnabled={true}
  injectedJavaScriptObject={{ customValue: 'myCustomValue' }}
  injectedJavaScript={`window.testMessage = "hello world"`}
/>
```

It'd be awesome if this could be the first steps towards having a decently usable lexical version on react native

**Reply by strdr4605** [reply]
It is possible if you pass the whole lexical state as a variable and use it in editor config
<img width="1760" alt="image" src="https://github.com/strdr4605/strdr4605.github.io/assets/16056918/106dd3a6-8ea5-4e21-9921-13f4ac3b3c39">
